src/powershell/Public/Remove-FinOpsHub.ps1

```diff
@@ -0,0 +1,113 @@
+
+    .PARAMETER ResourceGroupName
+    Optional when specifying Name. Resource Group Name for the FinOps Hub.
+
+    .PARAMETER InputObject
+    Required when specifying InputObject. Expected object is the output of Get-FinOpsHub.
+
+    .PARAMETER KeepStorageAccount
+    Optional. Indicates that the storage account associated with the FinOps Hub should be retained.
+
+    .EXAMPLE
+    Remove-FinOpsHub -Name MyHub -ResourceGroupName MyRG -KeepStorageAccount
+
+    Deletes a FinOps Hub named MyHub and deletes all associated resource except the storagea ccount.
+#>
+
+function Remove-FinOpsHub
+{
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'High')]
+    param
+    (
+        [Parameter(Mandatory = $true, ParameterSetName = 'Name')]
+        [ValidateNotNullOrEmpty ()]
+        [string]
+        $Name,
+
+        [Parameter(ParameterSetName = 'Name')]
+        [string]
+        $ResourceGroupName,
+
+        [Parameter(Mandatory = $true, ParameterSetName = 'Object')]
+        [ValidateNotNullOrEmpty ()]
+        [psobject]
+        $InputObject,
+
+        [Parameter(ParameterSetName = 'Name')]
+        [Parameter(ParameterSetName = 'Object')]
+        [switch]
+        $KeepStorageAccount,
+
+        [Parameter(ParameterSetName = 'Name')]
+        [Parameter(ParameterSetName = 'Object')]
+        [switch]
+        $Force
+    )
+
+    $context = Get-AzContext
+    if (-not $context)
+    {
+        throw $script:localizedData.ContextNotFound
+    }
+
+    try
+    {
+
+        $fhO = $null
+
+        if ($PSCmdlet.ParameterSetName -eq 'Name')
+        {
+            if (-not [string]::IsNullOrEmpty($ResourceGroupName))
+            {
+                $fhO = Get-FinOpsHub -Name $Name -ResourceGroupName $ResourceGroupName
+            }
+            else
+            {
+                $fhO = Get-FinOpsHub -Name $Name
+            }
+        }
+        else
+        {
+            $fhO = $InputObject
+        }
+
+        # TODO: $Name is not available if InputObject provided
+        if (-not $fhO)
+        {
+            throw $script:localizedData.FinOpsHubNotFound -f $Name
+        }
+
+        # Extract unique identifier from Key Vault name
+        $kv = $fhO.Resources | Where-Object ResourceType -eq "Microsoft.KeyVault/vaults"
+        $uniqueId = $kv[0].Substring($kv[0].LastIndexOf("-") + 1)
+
+        $resources = $null
+        if ($KeepStorageAccount)
+        {
+            $resources = Get-AzResource -ResourceGroupName $ResourceGroup | Where-Object Name -like "*$uniqueId*" | Where-Object ResourceType -ne "Microsoft.Storage/storageAccounts"
+        }
+        else {
+            $resources = Get-AzResource -ResourceGroupName $ResourceGroup | Where-Object Name -like "*$uniqueId*"
+        }
+
+        #$resources | ft ResourceId, Name, ResourceType
+
+        if ($PSCmdlet.ShouldProcess($Name, 'DeleteFinOpsHub'))
+        {
+            $resources | Remove-AzResource -Force:$Force -AsJob
```

Comment:
I would make AsJob a parameter because it might be confusing to the user if it returns immediately while its still running in the background. You may also need to cleanup and return the output of the job

Reply:
Removed -AsJob :)